integration_tests/base_routes.py

```diff
@@ -449,6 +447,19 @@ async def async_template_render():
     return template
 
 
+# add route and it's function for url_for for template for integration testing
+
+
+@app.get("/sync/url_for")
+def sync_url_for():
+    return jinja_template.url_for("/sync/url_for")
+
+
+@app.get("/async/url_for")
+async def async_url_for():
+    return jinja_template.url_for("/async/url_for")
```

Comment:
@carrycooldude , you are supposed to return the `render_template` method and then test the `url_for` in the rendered html.

Reply:
@sansyrox like this 
```
jinja_template.add_template_global(url_for)
```